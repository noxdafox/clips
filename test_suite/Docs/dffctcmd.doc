;;;************************************************************;;; DEFFACTS COMMAND TESTING;;;;;; The file dffctcmd.tst tests the commands described in the;;; BPG in section 11.6 (deffacts commands).;;;;;; To test, execute the command (batch "dffctcmd.tst").;;; When the test is complete, check the file dffctcmd.rsl;;; for the results (differences between expected and ;;; actual output are listed).;;;************************************************************