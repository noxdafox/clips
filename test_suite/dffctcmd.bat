(clear)
(deftemplate foo (field x) (multifield y))
(deffacts d1)
(deffacts d2 (a))
(deffacts d3 (a) (b) (c))
(deffacts d4 (foo (x 3) (y)))
(deffacts d5 (a) (foo (y a) (x 3)))
(deffacts d6 (foo (y a b)) (b) (foo (x 3)) (d))
(deffacts *)
(list-deffacts d6)
(list-deffacts)
(ppdeffacts d1)
(ppdeffacts d2)
(ppdeffacts d3)
(ppdeffacts d4)
(ppdeffacts d5)
(ppdeffacts d6)
(undeffacts)
(undeffacts 10)
(undeffacts bad-name)
(undeffacts d6 10)
(list-deffacts)
(undeffacts d1)
(undeffacts d2)
(undeffacts d3)
(list-deffacts)
(undeffacts *)
(list-deffacts)
(undeffacts *)
(list-deffacts)
(clear) ; Some new 6.0 features
(deffacts BOGUS::d5)
(defmodule FOO)
(deffacts FOO::d1)
(deffacts FOO::d2)
(defmodule BAR)
(deffacts BAR::d2)
(deffacts BAR::d3)
(list-deffacts BOGUS)
(list-deffacts FOO 7)
(list-deffacts FOO)
(list-deffacts BAR)
(list-deffacts)
(list-deffacts *)
(get-deffacts-list 7)
(get-deffacts-list BOGUS 7)
(get-deffacts-list)
(get-deffacts-list BOGUS)
(get-deffacts-list FOO)
(get-deffacts-list BAR)
(get-deffacts-list *)
(ppdeffacts d1)
(ppdeffacts d2)
(ppdeffacts d3)
(ppdeffacts BOGUS::d2)
(ppdeffacts FOO::d1)
(ppdeffacts FOO::d2)
(ppdeffacts FOO::d3)
(ppdeffacts BAR::d1)
(ppdeffacts BAR::d2)
(ppdeffacts BAR::d3)
(deffacts-module)
(deffacts-module bogus)
(deffacts-module d3 7)
(deffacts-module d2)
(deffacts-module d3)
(deffacts-module FOO::d1)
(deffacts-module FOO::d2)
(deffacts-module BAR::d2)
(deffacts-module BAR::d3)
(undeffacts BOGUS::d2)
(undeffacts FOO::d2)
(list-deffacts *)
(undeffacts BAR::d2)
(list-deffacts *)
(undeffacts *)
(list-deffacts *)
