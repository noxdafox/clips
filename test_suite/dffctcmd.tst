(unwatch all)
(clear)
(dribble-on "Actual//dffctcmd.out")
(batch "dffctcmd.bat")
(dribble-off)
(clear)
(open "Results//dffctcmd.rsl" dffctcmd "w")
(load "compline.clp")
(printout dffctcmd "dffctcmd.bat differences are as follows:" crlf)
(compare-files "Expected//dffctcmd.out" "Actual//dffctcmd.out" dffctcmd)
(close dffctcmd)
