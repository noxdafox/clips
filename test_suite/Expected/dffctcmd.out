TRUE
CLIPS> (batch "dffctcmd.bat")
TRUE
CLIPS> (clear)
CLIPS> (deftemplate foo (field x) (multifield y))
CLIPS> (deffacts d1)
CLIPS> (deffacts d2 (a))
CLIPS> (deffacts d3 (a) (b) (c))
CLIPS> (deffacts d4 (foo (x 3) (y)))
CLIPS> (deffacts d5 (a) (foo (y a) (x 3)))
CLIPS> (deffacts d6 (foo (y a b)) (b) (foo (x 3)) (d))
CLIPS> (deffacts *)
CLIPS> (list-deffacts d6)
[ARGACCES5] Function list-deffacts expected argument #1 to be of type defmodule name
CLIPS> (list-deffacts)
initial-fact
d1
d2
d3
d4
d5
d6
*
For a total of 8 deffacts.
CLIPS> (ppdeffacts d1)
(deffacts MAIN::d1)
CLIPS> (ppdeffacts d2)
(deffacts MAIN::d2
   (a))
CLIPS> (ppdeffacts d3)
(deffacts MAIN::d3
   (a)
   (b)
   (c))
CLIPS> (ppdeffacts d4)
(deffacts MAIN::d4
   (foo (x 3) (y)))
CLIPS> (ppdeffacts d5)
(deffacts MAIN::d5
   (a)
   (foo (y a) (x 3)))
CLIPS> (ppdeffacts d6)
(deffacts MAIN::d6
   (foo (y a b))
   (b)
   (foo (x 3))
   (d))
CLIPS> (undeffacts)
[ARGACCES4] Function undeffacts expected exactly 1 argument(s)
CLIPS> (undeffacts 10)
[ARGACCES5] Function undeffacts expected argument #1 to be of type symbol
CLIPS> (undeffacts bad-name)
[PRNTUTIL1] Unable to find deffacts bad-name.
CLIPS> (undeffacts d6 10)
[ARGACCES4] Function undeffacts expected exactly 1 argument(s)
CLIPS> (list-deffacts)
initial-fact
d1
d2
d3
d4
d5
d6
*
For a total of 8 deffacts.
CLIPS> (undeffacts d1)
CLIPS> (undeffacts d2)
CLIPS> (undeffacts d3)
CLIPS> (list-deffacts)
initial-fact
d4
d5
d6
*
For a total of 5 deffacts.
CLIPS> (undeffacts *)
CLIPS> (list-deffacts)
initial-fact
d4
d5
d6
For a total of 4 deffacts.
CLIPS> (undeffacts *)
CLIPS> (list-deffacts)
CLIPS> (clear) ; Some new 6.0 features
CLIPS> (deffacts BOGUS::d5)
[PRNTUTIL1] Unable to find defmodule BOGUS.

ERROR:
(deffacts BOGUS::d5
CLIPS> (defmodule FOO)
CLIPS> (deffacts FOO::d1)
CLIPS> (deffacts FOO::d2)
CLIPS> (defmodule BAR)
CLIPS> (deffacts BAR::d2)
CLIPS> (deffacts BAR::d3)
CLIPS> (list-deffacts BOGUS)
[ARGACCES5] Function list-deffacts expected argument #1 to be of type defmodule name
CLIPS> (list-deffacts FOO 7)
[ARGACCES4] Function list-deffacts expected no more than 1 argument(s)
CLIPS> (list-deffacts FOO)
d1
d2
For a total of 2 deffacts.
CLIPS> (list-deffacts BAR)
d2
d3
For a total of 2 deffacts.
CLIPS> (list-deffacts)
d2
d3
For a total of 2 deffacts.
CLIPS> (list-deffacts *)
MAIN:
   initial-fact
FOO:
   d1
   d2
BAR:
   d2
   d3
For a total of 5 deffacts.
CLIPS> (get-deffacts-list 7)
[ARGACCES5] Function get-deffacts-list expected argument #1 to be of type symbol
CLIPS> (get-deffacts-list BOGUS 7)
[ARGACCES4] Function get-deffacts-list expected no more than 1 argument(s)
CLIPS> (get-deffacts-list)
(d2 d3)
CLIPS> (get-deffacts-list BOGUS)
[ARGACCES5] Function get-deffacts-list expected argument #1 to be of type defmodule name
()
CLIPS> (get-deffacts-list FOO)
(d1 d2)
CLIPS> (get-deffacts-list BAR)
(d2 d3)
CLIPS> (get-deffacts-list *)
(MAIN::initial-fact FOO::d1 FOO::d2 BAR::d2 BAR::d3)
CLIPS> (ppdeffacts d1)
[PRNTUTIL1] Unable to find deffacts d1.
CLIPS> (ppdeffacts d2)
(deffacts BAR::d2)
CLIPS> (ppdeffacts d3)
(deffacts BAR::d3)
CLIPS> (ppdeffacts BOGUS::d2)
[PRNTUTIL1] Unable to find deffacts BOGUS::d2.
CLIPS> (ppdeffacts FOO::d1)
(deffacts FOO::d1)
CLIPS> (ppdeffacts FOO::d2)
(deffacts FOO::d2)
CLIPS> (ppdeffacts FOO::d3)
[PRNTUTIL1] Unable to find deffacts FOO::d3.
CLIPS> (ppdeffacts BAR::d1)
[PRNTUTIL1] Unable to find deffacts BAR::d1.
CLIPS> (ppdeffacts BAR::d2)
(deffacts BAR::d2)
CLIPS> (ppdeffacts BAR::d3)
(deffacts BAR::d3)
CLIPS> (deffacts-module)
[ARGACCES4] Function deffacts-module expected exactly 1 argument(s)
CLIPS> (deffacts-module bogus)
[PRNTUTIL1] Unable to find deffacts bogus.
FALSE
CLIPS> (deffacts-module d3 7)
[ARGACCES4] Function deffacts-module expected exactly 1 argument(s)
CLIPS> (deffacts-module d2)
BAR
CLIPS> (deffacts-module d3)
BAR
CLIPS> (deffacts-module FOO::d1)
FOO
CLIPS> (deffacts-module FOO::d2)
FOO
CLIPS> (deffacts-module BAR::d2)
BAR
CLIPS> (deffacts-module BAR::d3)
BAR
CLIPS> (undeffacts BOGUS::d2)
[PRNTUTIL1] Unable to find deffacts BOGUS::d2.
CLIPS> (undeffacts FOO::d2)
CLIPS> (list-deffacts *)
MAIN:
   initial-fact
FOO:
   d1
BAR:
   d2
   d3
For a total of 4 deffacts.
CLIPS> (undeffacts BAR::d2)
CLIPS> (list-deffacts *)
MAIN:
   initial-fact
FOO:
   d1
BAR:
   d3
For a total of 3 deffacts.
CLIPS> (undeffacts *)
CLIPS> (list-deffacts *)
MAIN:
   initial-fact
FOO:
   d1
BAR:
For a total of 2 deffacts.
CLIPS> (dribble-off)
